tests/unit/archive_base_test.py

```diff
@@ -166,6 +166,33 @@ async def get_resource(self, i):
         assert download_paths[resource.partitions["idx"]] == name
 
 
+@pytest.mark.asyncio
+async def test_fail_fast(bad_zipfile):
```

Comment:
We already have a `bad_zipfile` fixture in this unit test file, which creates in ill-formatted zipfile and returns the path. I've reused that here and created a `MockArchiver` that makes the base class think it just downloaded this bad zipfile, and when we pass the `fail_fast` option to `download_all_resources` it will raise an error immediately when it encounters it.